File: app/src/main/java/com/afoxxvi/alopex/ui/dialog/TextDialog.java
package com.afoxxvi.alopex.ui.dialog;

import android.content.Context;
import android.text.SpannableStringBuilder;
import android.text.method.ScrollingMovementMethod;
import android.view.LayoutInflater;

import com.afoxxvi.alopex.databinding.DialogSingleTextBinding;

public abstract class TextDialog extends BaseDialog {
    private final DialogSingleTextBinding binding;

    public TextDialog(Context context, String header, SpannableStringBuilder text) {
        super(context, header);
        binding = DialogSingleTextBinding.inflate(LayoutInflater.from(context));
        setContent(binding.getRoot());
        binding.content.setText(text);
        binding.content.setMovementMethod(ScrollingMovementMethod.getInstance());
    }
}

File: app/src/main/java/com/afoxxvi/alopex/ui/dialog/TextDialog.kt
package com.afoxxvi.alopex.ui.dialog

import android.content.Context
import android.text.SpannableStringBuilder
import android.text.method.ScrollingMovementMethod
import android.view.LayoutInflater
import com.afoxxvi.alopex.databinding.DialogSingleTextBinding

abstract class TextDialog(context: Context, header: String?, text: SpannableStringBuilder?) : BaseDialog(context, header) {
    private val binding: DialogSingleTextBinding

    init {
        binding = DialogSingleTextBinding.inflate(LayoutInflater.from(context))
        setContent(binding.root)
        binding.content.text = text
        binding.content.movementMethod = ScrollingMovementMethod.getInstance()
    }
}